include/proteus/Frontend/DispatcherHost.hpp

```diff
@@ -70,7 +71,11 @@ class DispatcherHost : public Dispatcher {
 
   void registerDynamicLibrary(HashT HashValue,
                               const SmallString<128> &Path) override {
-    ObjectCache.storeDynamicLibrary(HashValue, Path);
+    auto Buf = MemoryBuffer::getFile(Path);
```

Comment:
Replace with `MemoryBuffer::getFileAsStream`, it is more robust (check the API docs)